tools/driver/p4c_src/main.py

```diff
@@ -184,20 +192,30 @@ def main():
         print display_supported_targets(cfg)
         sys.exit(0)
 
+    # When using --help-* options, we don't necessarily need to pass an input file
+    # However, by default the driver checks the input and fails if it does not exist.
+    # Also, loading the backend configuration files requires a source file to be set,
+    # so in that case, we set the input to dummy.p4 and expect that the backend itself
+    # will do its printing and ignore the input file.
+    # If not, the compiler will error out anyway.
+    checkInput = True if not opts.help_pragmas else False
```

Comment:
Why not just `checkInput = not opts.help_pragmas`?

Reply:
good point.